Set the @nearby_competitions object if creation fails.

pedrosino opened the pull request:
Fixes #1016 (again)

viroulep commented on WcaOnRails/spec/features/competition_management_spec.rb:
```diff
@@ -14,6 +14,22 @@
+      visit "/competitions/new"
+      click_button "Create Competition"
+
+      expect(page).to have_text("Name must end with a year")
```
Very naive question here: shouldn't this be a separate scenario?
Like scenario 1 is creating a valid competition, scenario 2 is attempting to create an invalid competition?

pedrosino replied:
I guess you're right. I don't know much about this scenarios thing.

viroulep commented on WcaOnRails/spec/features/competition_management_spec.rb:
```diff
@@ -14,6 +14,22 @@
+      expect(page).to have_text("Name must end with a year")
+    end
+
+    scenario "User clones an existing competition" do
```
Maybe add in the description that you are testing the validation part? (like add "with validation errors" at the end)

pedrosino replied:
Right.

pedrosino commented:
>During the review I noticed that we have both `manage_competition_spec.rb` and `competition_management_spec.rb`, and there is at least one duplicate scenario in both file: is there a fundamental difference between them, or could we merge them up somehow?

I can't answer this, as I didn't create any of those files. I just added a test where it felt right :)

pedrosino commented:
Pedro said
>I can't answer this, as I didn't create any of those files. I just added a test where it felt right :)

Actually I added a test to the file @jfly  told me to. Looks like he's responsible for the redundancy: https://github.com/thewca/worldcubeassociation.org/commit/ecc373a23b501e8742f5294ad67723a69596e79e 
(thanks @jonatanklosko for finding this)

jfly commented:
> Actually I added a test to the file @jfly told me to. Looks like he's responsible for the redundancy: ecc373a
(thanks @jonatanklosko for finding this)

Lol! I did not realize we had both of those. @pedrosino, could you combine competition_management_spec.rb and manage_competition_spec.rb into one file? You can choose whichever name you prefer =)

pedrosino commented:
I did some adjusting, but I'm not very sure if things look good.

viroulep commented:
LGTM :)

pedrosino commented:
I'm merging this to solve the issue. We can improve the tests later if something looks wrong.
